### PR TITLE
React professor page (READY TO MERGE)

### PR DESCRIPTION
**Pull request by bradknaysi**
- still have to do outer `subdepartment_chunk` loop and inner `courses` loop. ✅ 

- Need to get course stats from `@course.stats`, which is difficult as it requires matching with correct `@professor` ✅ 

- Get "LAST TAUGHT" working ✅ 

- Add `@professor.email` ✅ 

- Make UI browser window shrink and stretch resistant (Responsive Web Design) ✅ 

## This is how my new UI looks...
PR_411_LATEST

### CODE REVIEW

**Comment by brian-yu** [comment]
@bradknaysi Functionality looks awesome but I think we should look into getting the design and styling 100% before we launch it. @saikonuri thoughts? I'll also do an in depth code review in a bit when I have more time.

**Comment by bradknaysi** [comment]
@brian-yu What do you think should be changed?

File: app/assets/stylesheets/professor.scss [hunk 1]
```diff
@@ -6,58 +6,105 @@
...
-}
-
-#professor-courses {
-	.course-panel {
-		margin-left: 10px;
-		margin-right: 10px;
-	}
-}
\ No newline at end of file
+  margin-top:    0px;
```
Comment (brian-yu) [inline app/assets/stylesheets/professor.scss]: Maybe we should put this in `app/javascript/components/professors/ProfessorPage.css`? Then we can import the styles in React.

File: app/javascript/components/professors/ProfessorPage.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        </div>
+      );
+    });
+
+    return (subdepartments);
+  }
+
+
+  render() {
+    const professorHeader = this.buildProfessorHeader.call();
```
Comment (brian-yu) [inline app/javascript/components/professors/ProfessorPage.js]: I believe we can just use `const professorHeader = this.buildProfessorHeader();`

File: app/javascript/components/professors/ProfessorPage.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    const subdepartments = this.props.course_groups.map((subdepartment) => {
+      const courses = subdepartment[1].map((course) =>
+        <div key={course.course.id} className="row course-box">
+          <div className="row button-row">
+            <button className=" btn btn-primary course-button" type="button" onClick={this.handleCourseClick.bind(this, course)}>
+              {course.course.title} ({course.course.course_number})
+            </button>
+          </div>
+          <div className="row ratings-row">
+            {this.buildCourseDetailsBlock.call(this, "RATING", course.course_stats.rating)}
```
Comment (brian-yu) [inline app/javascript/components/professors/ProfessorPage.js]: `this. buildCourseDetailsBlock("RATING", course.course_stats.rating)`

**Pull request merged**